pages/login/index.vue

```diff
@@ -46,24 +48,49 @@
 </template>
 
 <script>
-import { useContext } from '@nuxtjs/composition-api'
+import { useContext, reactive, toRefs } from '@nuxtjs/composition-api'
 import { defineComponent } from '@vue/composition-api'
 
 export default defineComponent({
+  middleware: 'auth',
   setup() {
-    const { $toast } = useContext()
+    const state = reactive({
+      username: '',
+      password: '',
+    })
+    const { redirect, $axios, store, $toast } = useContext()
+
+    const loginFieldsAreValid = () => {
+      if (state.username === '' || state.password === '') {
```

Comment:
A better feedback can be to tell the problem ie the fields cannot be empty to the user